src/player/Player.php

```diff
@@ -1585,14 +1585,13 @@ public function pickBlock(Vector3 $pos, bool $addTileNBT) : bool{
 			return true;
 		}
 
-		$item = $block->getPickedItem($addTileNBT);
-
-		$ev = new PlayerBlockPickEvent($this, $block, $item);
+		$ev = new PlayerBlockPickEvent($this, $block, $block->getPickedItem($addTileNBT));
+		$ev->call();
```

Comment:
Why is this moved? The code below it needs to run before the event is called.

Reply:
The code below cancels the event before calling it if the item can't be added to the inventory. Your change breaks that behaviour.